models/streamitem.go

```diff
@@ -0,0 +1,110 @@
+package models
+
+import (
+	"math"
+
+	"github.com/bwmarrin/snowflake"
+	"github.com/jinzhu/gorm"
+	"github.com/meowpub/meow/config"
+	"github.com/meowpub/meow/lib"
+)
+
+//go:generate mockgen -package=models -source=streamitem.go -destination=streamitem.mock.go
+
+// StreamItem represents a single item in a string
+type StreamItem struct {
+	ItemID   snowflake.ID `json:"item_id"`
+	StreamID snowflake.ID `json:"stream_id"`
+	EntityID snowflake.ID `json:"entity_id"`
+}
+
+type Direction int
+
+const (
+	Before Direction = -1
+	After  Direction = 1
+
+	Beginning snowflake.ID = 0
+	End       snowflake.ID = math.MaxInt64
+)
+
+type StreamItemStore interface {
+	// GetItems returns count items from the specified stream, starting
+	// after/before the specified item, in the specified direction. Pass "Beginning"/"End" to
+	// start at the beginning or end of the stream
+	GetItems(streamID snowflake.ID, startID snowflake.ID, direction Direction, count uint) ([]StreamItem, error)
+
+	// GetItem returns a specific stream item
+	GetItem(itemID snowflake.ID) (*StreamItem, error)
+
+	// GetItemByEntityId returns a specific stream item by stream and entry ID
+	GetItemByEntityID(streamID snowflake.ID, entityID snowflake.ID) (*StreamItem, error)
+
+	// TryInsertItem attempts to insert the specified entity ID into the stream.
+	// Returns:
+	//   * If the entry was inserted
+	//   * The stream item (regardless)
+	TryInsertItem(stream snowflake.ID, entityId snowflake.ID) (bool, *StreamItem, error)
```

Comment:
The `*StreamItem` should be the first return value, that'd be consistent with the stdlib